### ISSUE TITLE
executeAsynchronously discards constructed std::thread

### ISSUE THREAD

**Issue by asklar** [facebook/react-native#31088]
https://github.com/facebook/react-native/blob/6e6443afd04a847ef23fb6254a84e48c70b45896/ReactCommon/runtimeexecutor/ReactCommon/RuntimeExecutor.h#L42-L44

the std::thread ctor is nodiscard in the latest MSVC; std::thread doesn't support fire-and-forget, in order to do that you'd have to call .detach() on it
